### PR TITLE
fix: ignore NA/NaN types for feature group

### PR DESCRIPTION
**Pull request by ahsan-z-khan**
*Issue #, if available:*
#2110 
*Description of changes:*
added support to ignore NA/NaN for feature group
*Testing done:*
tox tests/unit --parallel=all
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by cst-care** [comment]
Hi Ahsan,
You added the line
`if str(row[index]) != "NaN" or str(row[index]) != "NA"`
Would it be possible to also take into account "None" and "NaT"?
It might be even easier to just use the pandas function as it includes all of the above:
`if pd.notnull(row[index])`
Thanks

File: src/sagemaker/feature_store/feature_group.py [hunk 1]
```diff
@@ -185,6 +185,7 @@ def _ingest_single_batch(
                     feature_name=data_frame.columns[index], value_as_string=str(row[index])
                 )
                 for index in range(len(row))
```
Comment (icywang86rui) [inline src/sagemaker/feature_store/feature_group.py:187]:
```suggestion
                for index in range(len(row)) if pd.notna(row[index])
```

Comment (icywang86rui) [reply src/sagemaker/feature_store/feature_group.py:187]: we need tests for this too

**Review by icywang86rui — CHANGES_REQUESTED** [review]

**Comment by pkavousi** [comment]
Hello. Thanks for the above discussion and solutions. Am I correct to assume that the below record can't be ingested into a feature group?

[{'FeatureName': 'ji', 'ValueAsString': '8829a094'}, {'FeatureName': 'time', 'ValueAsString': '2020-08-27T13:00:00Z'}, {'FeatureName': 't2', 'ValueAsString': '289.26111111111106'}, {'FeatureName': 're', 'ValueAsString': '86'}, {'FeatureName': 'pwat', 'ValueAsString': '0.9609375'}, {'FeatureName': 'li700', 'ValueAsString': '3'}, {'FeatureName': 'c', 'ValueAsString': '0'}, {'FeatureName': 'd', 'ValueAsString': '0'}, {'FeatureName': 'x', 'ValueAsString': '0'}, {'FeatureName': 'y', 'ValueAsString': '0.0'}, {'FeatureName': 'z', 'ValueAsString': 'None'}]

I get this error:
Attempted to parse the feature value for the feature named [z] into a FeatureValue of type Fractional. The provided value must be within the range of a double precision floating point number defined by the IEEE 754 standard. The input format can be in either decimal form or scientific notation.

Is there anyway to ingest this into a feature group?

**Pull request closed without merging**